libs/vectorise/include/vectorise/fixed_point/fixed_point.hpp

```diff
@@ -475,9 +495,29 @@ class FixedPoint : public BaseFixedpointType
   static constexpr FixedPoint         CosPi2(FixedPoint const &r);
   static constexpr FixedPoint<16, 16> SinApproxPi4(FixedPoint<16, 16> const &r);
   static constexpr FixedPoint<32, 32> SinApproxPi4(FixedPoint<32, 32> const &r);
+  static constexpr FixedPoint<64, 64> SinApproxPi4(FixedPoint<64, 64> const &r);
   static constexpr FixedPoint         CosApproxPi4(FixedPoint const &r);
 };
 
+template <typename T>
+static inline bool IsMinusZero(T n);
+
+template <>
+inline bool IsMinusZero<float>(float const n)
+{
+  // Have to dereference the pointer as char const *, else we get a strict-aliasing error
+  uint32_t n_int = *(reinterpret_cast<uint8_t const *>(&n));
+  return (n_int == 0x80000000);
```

Comment:
Even putting aside that C++ `float` is not necessarily an IEEE-754, endianness is ignored here.

Reply:
currently we do not even test on big-endian systems. And this is not float, neither it is IEEE754 compliant. It is our implementation of a fixed point.